### PR TITLE
Create mnist_cnn_features_level_fgsm.py

### PR DESCRIPTION
**Pull request by step8p**
Example of how to perturb a middle layer of a model instead of perturbing the input one

# Description

The example is based on the "mnist_cnn_fgsm.py" one. The added parts allow perturbing a middle layer. No new import added. NOTE: the "clip_values" array has been removed since it is not really sure what should be the bounds of a middle layer perturbation, but it could be inserted if needed (e.g. if you want that the perturbed values are still > 0 since the previous layer uses a relu).

Fixes # (issue)

## Type of change

Please check all relevant options.

- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: examples/mnist_cnn_features_level_fgsm.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""Trains a convolutional neural network on the MNIST dataset, then attacks it with the FGSM attack."""
```
Comment (ririnicolae) [inline examples/mnist_cnn_features_level_fgsm.py]: Could you please update this comment to reflect that the example attacks an inner layer?

File: examples/mnist_cnn_features_level_fgsm.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+acc = np.sum(preds == np.argmax(y_test, axis=1)) / y_test.shape[0]
+print("\nTest accuracy: %.2f%%" % (acc * 100))
+
+# Craft adversarial samples with FGSM
+epsilon = .1  # Maximum perturbation
+adv_crafter = FastGradientMethod(classifier, eps=epsilon)
+x_test_adv = adv_crafter.generate(x=x_test_inner)
+
+# Evaluate the classifier on the adversarial examples
+preds = np.argmax(classifier.predict(x_test_adv), axis=1)
```
Comment (ririnicolae) [inline examples/mnist_cnn_features_level_fgsm.py]: I wonder if here you should evaluate `x_test_adv` on the full lclassifier or only on the second half on which the samples were produced? `x_test_adv` is `x_test_inner` plus some epsilon quantity, so it's not in the input space, but in the same feature space as `x_test_inner`. What do you think?

**Comment by ririnicolae** [comment]
Hey, @step8p! Any updates on this PR?

**Comment by step8p** [comment]
Hello, sorry for late answering (caught in several deadlines). I'll work on them next week :)

File: examples/mnist_cnn_features_level_fgsm.py (hunk 1 above)
Comment (step8p) [reply examples/mnist_cnn_features_level_fgsm.py]: done

**Comment by step8p** [comment]
thank you for your work!! This toolbox is very useful :)

**Comment by beat-buesser** [comment]
Thank you @step8p and @ririnicolae ! Great work.

**Pull request merged**